### DISCUSSION TITLE
Question: Is it possible to force diff folder for opengrok lock file?

### DISCUSSION THREAD

**Discussion by mmmarq** [oracle/opengrok#4361] [General]
Hello all,

I'm in a try re-design my opengrok index/webapp architecture. Since my servers are suffering to run both index/webapp at the same time, I'm trying to move index process to a diff server, where I plan to run parallel index for diff code bases (not projects). But opengrok.jar has no documented option to change opengrok-indexer.lock file location. So I cannot run index process in parallel.
I have tried "--java_opts=-Djava.io.tmpdir=/new/tmp/path" but it had no effect.
Does anybody knows a way to change it?
Thanks in advance

**Comment by vladak** [comment]
Do you run the indexer with the `--lock` option ? How does the error message look like ? It's not clear to me where does the `opengrok-indexer.lock` come from.

**Comment by mmmarq** [comment]
Hello @vladak 

No, I did not use `--lock` option: A sample command is:

```
/tools/jre/bin/java -Djava.io.tmpdir=/opengrok/temp -Dorg.opengrok.indexer.history.git=/tools/git/bin/git -Xmx32g -server -jar /tools/opengrok/lib/opengrok.jar --ctags /tools/ctags/ctags --dataRoot /opengrok/data -i d:.repo -i d:.git -H --depth 50 --source /opengrok/src_root --project -R /opengrok/etc/configuration.xml <project_name> <project_name>
```
And lock file is created at:
`opengrok-indexer.lock`

Lock file creation can be seen at the first line of the command output:
```
Unable to determine Universal CTags command
DEBUG:filelock:Attempting to acquire lock 140468404938208 on /tmp/opengrok-indexer.lock
DEBUG:filelock:Lock 140468404938208 acquired on /tmp/opengrok-indexer.lock
May 29, 2023 12:18:47 PM org.opengrok.indexer.configuration.Configuration read
```
Thanks

**Reply by vladak** [reply]
This lockfile is created by the Python tools, which means setting Java system properties will not help there; the real indexer is running as a child process of the Python wrapper. Which Python tool do you use to run the indexer ?

**Comment by mmmarq** [comment]
Oh, really? Danm! My bad!
Java command is called by `opengrok-tools/bin/opengrok-indexer` wrapper. I think I'll be able to change it vi regular env var (like TEMP and TMP) so.

**Reply by vladak** [reply]
Yep, it uses the standard `tempfile.gettempdir()` to get the directory for the lock file. Possibly a no-lock mode can be added to the tool.

**Reply by mmmarq** [reply]
Oh no, let it as is. Locking by default saved my soul sometime in the past :-)

**Reply by mmmarq** [reply]
I'll test using env var and I will update this thread with the results

**Reply by vladak** [reply]
> Oh no, let it as is. Locking by default saved my soul sometime in the past :-)

The `opengrok-sync` has the `--nolock` option so it makes sense for the `opengrok-indexer` to have it as well. The locking will be enabled by default.

**Reply by mmmarq** [reply]
Hello @vladak,

Just by adding TEMP env var set before wraper script call changed lock file folder destination as expected:
`TEMP=${OPENGROK_PATH}/temp ${OPENGROK_PATH}/opengrok-tools/bin/opengrok-indexer --java ${JAVA_HOME}/bin/java --java_opts=-server --java_opts=-Xmx32g...`
```
DEBUG:filelock:Attempting to acquire lock 139792600333088 on /opengrok/temp/opengrok-indexer.lock
DEBUG:filelock:Lock 139792600333088 acquired on /opengrok/temp/opengrok-indexer.lock
```
Thank you very much for your time
